projects/js-packages/publicize-components/src/components/connection-icon/index.jsx

```diff
@@ -5,23 +5,37 @@ import PropTypes from 'prop-types';
 import './style.scss';
 
 const ConnectionIcon = props => {
-	const { id, serviceName, label, profilePicture } = props;
+	const { checked, serviceName, label, onClick, profilePicture } = props;
 	const [ hasDisplayPicture, setHasDisplayPicture ] = useState( !! profilePicture );
 
 	const onError = useCallback( () => setHasDisplayPicture( false ), [] );
 
+	const handleKeyDown = useCallback(
+		ev => {
+			if ( ev.keyCode === 13 ) {
+				onClick();
+			}
+		},
+		[ onClick ]
+	);
+
 	return (
-		<label htmlFor={ id } className="jetpack-publicize-connection-label">
-			<div className={ hasDisplayPicture ? 'components-connection-icon__picture' : '' }>
-				{ hasDisplayPicture && <img src={ profilePicture } alt={ label } onError={ onError } /> }
-				<SocialServiceIcon
-					serviceName={ 'instagram-business' === serviceName ? 'instagram' : serviceName }
-					className="jetpack-publicize-gutenberg-social-icon"
-					invert={ 'tumblr' === serviceName }
-				/>
-			</div>
-			<span className="jetpack-publicize-connection-label-copy">{ label }</span>
-		</label>
+		<div
```

Comment:
this just removes the label that was there for the toggle. And also adds some attrs to the div to be compliant with the a11n rules

Reply:
I've added the label back in as a title attribute, so that the accounts can be distinguished by hovering and getting a tooltip of the account name. We'll need to have a think about that for mobile.